testcases/manifests/deployment_with_strategy.yaml

```diff
@@ -0,0 +1,39 @@
+# {
+#     "status": "Success",
+#     "message": ""
+# }
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-service
+  namespace: default
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      service: my-service
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 0
+      maxSurge: 1
+  template:
+    metadata:
```

Comment:
Theres a bug in k8s's OpenAPI schemas right now that is defaulting creationTimestamp to empty string, which does not pass validation. We need to workaround it by applying a patch to the schemas.

would like to replace patterns like:

```json
"creationTimestamp": {
            "allOf": [
              {
                "$ref": "#/components/schemas/io.k8s.apimachinery.pkg.apis.meta.v1.Time"
              }
            ],
            "default": {},
            "description": "..."
          },
```

with this:

```json
  "creationTimestamp": {
            "allOf": [
              {
                "$ref": "#/components/schemas/io.k8s.apimachinery.pkg.apis.meta.v1.Time"
              }
            ],
            "default": null,
            "nullable": true,
            "description": "..."
          },
```

For now lets just manually patch `removeRefs` to insert `nullable` and a `null` default for optional fields that refer to a non-object.

Long term I think we should have a tool that scans a schema for defaults that don't pass validation.  It should be fixed in future versions of k8s but we should be able to find older instances of this problem.

Reply:
https://github.com/kubernetes-sigs/kubectl-validate/pull/60 fixes this